### PR TITLE
FAQ: why are flakes controversial

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
this is indeed a frequently asked question.

the answer is an attempt to dispassionately provide factual information.
we have ample evidence that Nix beginners get confused over the state of affairs.

### CODE REVIEW

File: source/recipes/faq.md [hunk 1]
```diff
@@ -2,6 +2,28 @@
...
+
+Originally proposed in [RFC 49](https://github.com/NixOS/rfcs/pull/49), flakes have been in development since 2019.
+Nix introduced the implementation as its first [experimental feature] in 2021.
+
+[experimental feature]: https://nixos.org/manual/nix/stable/contributing/experimental-features.html
+
+The subject is considered controversial among Nix users and developers in terms of design, development processes, and community governance.
+In particular:
+- The RFC was closed without conclusion, and some design and implementation issues are not yet resolved.
+  Examples include the notion of a global [flake registry], the [impossibility of parameterising flakes](https://github.com/NixOS/nix/issues/2861), and implementations of the new command line interface and flakes being [closely tied](https://discourse.nixos.org/t/2023-03-06-nix-team-meeting-minutes-38/26056#cli-stabilisation-announcement-draft-4).
```
Comment (fricklerhandwerk) [inline source/recipes/faq.md:15]:
```suggestion
  Examples include the notion of a global [flake registry], the [impossibility of parameterising flakes](https://github.com/NixOS/nix/issues/2861), and implementations of the new command line interface and flakes being [tightly coupled](https://discourse.nixos.org/t/2023-03-06-nix-team-meeting-minutes-38/26056#cli-stabilisation-announcement-draft-4).
```

**Review by brianmcgee — APPROVED** [review]
LGTM

File: source/recipes/faq.md [hunk 2]
```diff
@@ -2,6 +2,28 @@
...
+- The original implementation introduced [regressions](https://discourse.nixos.org/t/nix-2-4-and-what-s-next/16257) in the [Nix 2.4 release](https://nixos.org/manual/nix/stable/release-notes/rl-2.4.html).
+- New Nix users were and still are encouraged by various individuals to adopt flakes despite there being no concrete plan or timeline for stabilisation.
+
+[flake registry]: https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-registry.html
+
+This led to a situation where the stable interface was only sparsely maintained for multiple years, and repeatedly suffered breakages due to ongoing development.
+Meanwhile, the new interface was adopted widely enough for evolving its design without negatively affecting users to become very challenging.
+
+As of the [2022 community survey](https://discourse.nixos.org/t/2022-nix-survey-results/18983), more than half of the user base, a third of which were relative beginners, relied on experimental features.
+{term}`Nixpkgs` as a contrasting example, which, while featuring a `flake.nix` for compatibility, does not depend on Nix experimental features in its code base.
```
Comment (06kellyjac) [inline source/recipes/faq.md:25]:
```suggestion
{term}`Nixpkgs`, as a contrasting example, while featuring a `flake.nix` for compatibility, does not depend on Nix experimental features in its code base.
```

Or


```suggestion
As a contrasting example, while featuring a `flake.nix` for additional compatibility, {term}`Nixpkgs` does not depend on Nix experimental features in its code base.
```

The `, which,` makes it pretty hard to read IMO

**Review by 06kellyjac — APPROVED** [review]
LGTM :rocket:

**Comment by fricklerhandwerk** [comment]
@infinisil @roberth mind to take a look?

File: source/concepts/flakes.md [hunk 3]
```diff
@@ -40,3 +40,21 @@ Nix handles flakes differently than regular {term}`Nix file`s in the following w
...
+## Why are flakes controversial?
+
+Originally proposed in [RFC 49](https://github.com/NixOS/rfcs/pull/49), flakes have been in development since 2019.
+Nix introduced the implementation as its first [experimental feature] in 2021.
+
+The subject is considered controversial among Nix users and developers in terms of design, development processes, and community governance.
+In particular:
+- The RFC was closed without conclusion, and some design and implementation issues are not yet resolved.
+  Examples include the notion of a global [flake registry], the [impossibility of parameterising flakes](https://github.com/NixOS/nix/issues/2861), and implementations of the new command line interface and flakes being [closely tied](https://discourse.nixos.org/t/2023-03-06-nix-team-meeting-minutes-38/26056#cli-stabilisation-announcement-draft-4).
+- The original implementation introduced [regressions](https://discourse.nixos.org/t/nix-2-4-and-what-s-next/16257) in the [Nix 2.4 release](https://nixos.org/manual/nix/stable/release-notes/rl-2.4.html).
```
Comment (fricklerhandwerk) [inline source/concepts/flakes.md:53]:
```suggestion
- The original implementation introduced [regressions](https://discourse.nixos.org/t/nix-2-4-and-what-s-next/16257) in the [Nix 2.4 release](https://nixos.org/manual/nix/stable/release-notes/rl-2.4.html), breaking some stable functionality.
```

**Comment by fricklerhandwerk** [comment]
Had another review round with the documentation team.

**Pull request merged**